### PR TITLE
Fix boolean to string conversion in expr2java

### PR DESCRIPTION
**Pull request by peterschrammel**
Fixes diffblue/test-gen#206.

### CODE REVIEW

File: src/java_bytecode/expr2java.cpp [hunk 1]
```diff
@@ -195,9 +195,17 @@ std::string expr2javat::convert_constant(
   const constant_exprt &src,
   unsigned &precedence)
 {
-  if(src.type().id()==ID_bool)
+  if(src.type().id()==ID_c_bool)
+  {
+    mp_integer b;
+    to_integer(src, b);
+    if(b!=0)
```
Comment (tautschnig) [inline src/java_bytecode/expr2java.cpp]: Instead of `to_integer` you might just use `if(src.is_zero())`

File: src/java_bytecode/expr2java.cpp [hunk 2]
```diff
@@ -510,13 +517,8 @@ std::string expr2javat::convert(
...
     return '"'+MetaString(src.get_string(ID_value))+'"';
-  else if(src.id()==ID_constant && (type.id()==ID_bool || type.id()==ID_c_bool))
-  {
-    if(src.is_true())
-      return "true";
-    else
-      return "false";
-  }
+  else if(src.id()==ID_constant)
+    return convert_constant(to_constant_expr(src), precedence);
```
Comment (tautschnig) [inline src/java_bytecode/expr2java.cpp]: It's a somewhat wild guess: is this part of the change causing the regression test to fail (the regression test now has an extra set of `(`...`)` around `"java::D"`). Also I think that the `ID_nullptr` case should be moved to `convert_constant`.

Comment (peterschrammel) [inline src/java_bytecode/expr2java.cpp]: yes, indeed

**Comment by peterschrammel** [comment]
@tautschnig, all green.

**Pull request merged**